readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -121,15 +121,23 @@ def get_config_params(self):
                 )
             downloads = self.version.get_downloads(pretty=True)
         else:
-            if self.project.has_feature(Feature.ALL_VERSIONS_IN_HTML_CONTEXT):
+            versions = []
+            downloads = []
+            try:
                 versions = self.project.api_versions()
-            else:
-                versions = [
-                    v
-                    for v in self.project.api_versions()
-                    if v.privacy_level == PUBLIC
-                ]
-            downloads = api.version(self.version.pk).get()['downloads']
+                if not self.project.has_feature(Feature.ALL_VERSIONS_IN_HTML_CONTEXT):
+                    versions = [
+                        v
+                        for v in versions
+                        if v.privacy_level == PUBLIC
+                    ]
+                downloads = api.version(self.version.pk).get()['downloads']
+            except Timeout:
+                log.info(
```

Comment:
I'd probably send this to Sentry:

```suggestion
                log.exception(
```